onnxruntime/core/providers/webnn/builders/impl/gemm_op_builder.cc

```diff
@@ -162,6 +157,29 @@ bool GemmOpBuilder::IsOpSupportedImpl(const InitializedTensorSet& initializers,
     }
   }
 
+  if (op_type == "MatMul") {
+    if (a_shape.size() < 2 || b_shape.size() < 2) {
```

Comment:
For MatMul, do we have the code in ORT to coerce 1D tensors to 2D for WebNN, according to https://onnx.ai/onnx/operators/onnx__MatMul.html and https://numpy.org/doc/stable/reference/generated/numpy.matmul.html?

> - If both arguments are 2-D they are multiplied like conventional matrices.
> - If either argument is N-D, N > 2, it is treated as a stack of matrices residing in the last two indexes and broadcast accordingly.
> - If the first argument is 1-D, it is promoted to a matrix by prepending a 1 to its dimensions. After matrix multiplication the prepended 1 is removed.
> - If the second argument is 1-D, it is promoted to a matrix by appending a 1 to its dimensions. After matrix multiplication the appended 1 is removed.

If so, will that code no longer be reached because of this rejection?

Reply:
Good point, we should support 1-D input use `reshape`,  I suggest @zesongw to address this in next PR. As this PR will unblock most models testing on WebNN CPU.